cmake/SetupCodeChecks.cmake

```diff
@@ -294,3 +340,89 @@ macro(blt_add_uncrustify_target)
     endif()
 
 endmacro(blt_add_uncrustify_target)
+
+##------------------------------------------------------------------------------
+## blt_add_astyle_target( NAME              <Created Target Name>
+##                        MODIFY_FILES      [TRUE | FALSE (default)]
+##                        CFG_FILE          <AStyle Configuration File> 
+##                        PREPEND_FLAGS     <Additional Flags to AStyle>
+##                        APPEND_FLAGS      <Additional Flags to AStyle>
+##                        COMMENT           <Additional Comment for Target Invocation>
+##                        WORKING_DIRECTORY <Working Directory>
+##                        SRC_FILES         [FILE1 [FILE2 ...]] )
+##
+## Creates a new target with the given NAME for running astyle over the given SRC_FILES.
+##
+## MODIFY_FILES, if set to TRUE, modifies the files in place and adds the created target to
+## the style target. Otherwise the files are not modified and the created target is added
+## to the check target.  
+## Note: Setting MODIFY_FILES to FALSE is only supported in AStyle v2.05 or greater.
+##
+## CFG_FILE defines the astyle settings.
+##
+## PREPEND_FLAGS are additional flags added to the front of the astyle flags.
+##
+## APPEND_FLAGS are additional flags added to the end of the astyle flags.
+##
+## COMMENT is prepended to CMake's output for this target.
+##
+## WORKING_DIRECTORY is the directory in which astyle will be run. It defaults to 
+## the directory where this macro is called.
+##
+## SRC_FILES is a list of source files to style/check with astyle.
+##
+##------------------------------------------------------------------------------
+macro(blt_add_astyle_target)
+
+## parse the arguments to the macro
+set(options)
+set(singleValueArgs NAME MODIFY_FILES CFG_FILE COMMENT WORKING_DIRECTORY)
+set(multiValueArgs SRC_FILES PREPEND_FLAGS APPEND_FLAGS)
+
+cmake_parse_arguments(arg
+    "${options}" "${singleValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+# Check required parameters
+if(NOT DEFINED arg_NAME)
+    message(FATAL_ERROR "blt_add_astyle_target requires a NAME parameter")
+endif()
+
+if(NOT DEFINED arg_CFG_FILE)
+    message(FATAL_ERROR "blt_add_astyle_target requires a CFG_FILE parameter")
+endif()
+
+if(NOT DEFINED arg_SRC_FILES)
+    message(FATAL_ERROR "blt_add_astyle_target requires a SRC_FILES parameter")
+endif()
+
+if(NOT DEFINED arg_MODIFY_FILES)
+    set(arg_MODIFY_FILES FALSE)
+endif()
+
+if(DEFINED arg_WORKING_DIRECTORY)
+    set(_wd ${arg_WORKING_DIRECTORY})
+else()
+    set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
+
+if(${arg_MODIFY_FILES})
+    set(MODIFY_FILES_FLAG --suffix=none)
+else()
+    set(MODIFY_FILES_FLAG --formatted --dry-run)
+endif()
+
+add_custom_target(${arg_NAME}
+        COMMAND ${ASTYLE_EXECUTABLE} ${arg_PREPEND_FLAGS}
```

Comment:
Unlike uncrustify, astyle does not return non-zero when there are files that need to be formatted, so, as long as the config file is valid, these targets passes.

When there are files that need to be formatted, ``astyle`` outputs a line, per file that needs to be formatted of the form:
```cmake
Formatted <filename>
```

Should we add some code to run astyle and grep through the output for these lines so that the behavior is similar to that of our ``uncrustify`` targets?

Reply:
That is unfortunate. I know users want it to fail if files need to be formatted.

Reply:
If it's not too hard, we should try to parse and error. 
We will have to weigh with how brittle we feel parsing will be

Reply:
I added a wrapper script to run AStyle and parse the results. Both astyle_style and astyle_check targets now fail when there are improperly formatted source files.